Toggl.Core.Tests/UI/ViewModels/CalendarViewModelTests.cs

```diff
@@ -113,16 +150,366 @@ public void EmitsNewDateWhenCurrentlyVisiblePageChanges(int pageIndex, string ex
                 };
                 var observer = TestScheduler.CreateObserver<string>();
                 var viewModel = CreateViewModel();
+                var date = viewModel.IndexToDate(pageIndex);
                 viewModel.CurrentlyShownDateString.Subscribe(observer);
 
+                viewModel.CurrentlyShownDate.Accept(date);
                 TestScheduler.Start();
-                viewModel.CurrentlyVisiblePage.Accept(pageIndex);
+
+                observer.Values().Should().BeEquivalentTo(expectedResults);
+            }
+
+            [Theory, LogIfTooSlow]
+            [InlineData(4, 5, 2020, "Monday, May 4")]
+            [InlineData(30, 9, 1995, "Saturday, Sep 30")]
+            public void EmitsNewDateWhenAnItemInWeekViewIsTapped(int day, int month, int year, string expectedDateString)
+            {
+                var preferences = Substitute.For<IThreadSafePreferences>();
+                preferences.DateFormat.Returns(DateFormat.ValidDateFormats[0]);
+                DataSource.Preferences.Current.Returns(Observable.Return(preferences));
+                var expectedResults = new[]
+                {
+                    "Thursday, Jan 2",
+                    expectedDateString
+                };
+                var observer = TestScheduler.CreateObserver<string>();
+                var viewModel = CreateViewModel();
+                viewModel.CurrentlyShownDateString.Subscribe(observer);
+                var date = new DateTime(year, month, day);
+                var tappedWeekViewDayViewModel = new CalendarWeeklyViewDayViewModel(date, false, true);
+
+                viewModel.SelectDayFromWeekView.Execute(tappedWeekViewDayViewModel);
                 TestScheduler.Start();
 
                 observer.Values().Should().BeEquivalentTo(expectedResults);
             }
         }
 
+        public sealed class TheCurrentlyShownDateProperty : CalendarViewModelTest
+        {
+            private static readonly DateTimeOffset now = new DateTimeOffset(2020, 1, 2, 3, 4, 5, TimeSpan.Zero).ToLocalTime().Date;
+
+            public TheCurrentlyShownDateProperty()
+            {
+                TimeService.CurrentDateTime.Returns(now);
+            }
+
+            [Theory, LogIfTooSlow]
+            [InlineData(4, 5, 2020)]
+            [InlineData(30, 9, 1995)]
+            public void EmitsNewDateWhenAnItemInWeekViewIsSelected(int day, int month, int year)
+            {
+                var selectedDate = new DateTime(year, month, day);
+                var selectedDayViewModel = new CalendarWeeklyViewDayViewModel(selectedDate, false, true);
+                var preferences = Substitute.For<IThreadSafePreferences>();
+                preferences.DateFormat.Returns(DateFormat.ValidDateFormats[0]);
+                DataSource.Preferences.Current.Returns(Observable.Return(preferences));
+                var expectedResults = new[]
+                {
+                    new DateTime(2020, 1, 2),
```

Comment:
```suggestion
                    now.LocalDateTime,
```

Doing that ☝️ fixed the tests locally for me.